manifests/pulp.pp

```diff
@@ -97,10 +100,20 @@
   Optional[Stdlib::Absolutepath] $https_cert = undef,
   Optional[Stdlib::Absolutepath] $https_key = undef,
   Optional[Stdlib::Absolutepath] $https_ca_cert = undef,
+  Boolean $expose_mongodb = false,
+  Boolean $expose_mongodb_ipv6 = false,
 ) {
   include katello::params
   include certs
 
+  if !$pulp_manage_db and $pulp_expose_db {
+    fail("pulp_expose_db cannot be set to true if pulp_manage_db is false")
+  }
+
+  if !manage_mongodb and $expose_mongodb {
+    fail("expose_mongodb cannot be set to true if manage_mongodb is false")
+  }
```

Comment:
@ekohl Like this? This feels a bit weird having to define it twice due to having two sets of parameters.

Reply:
Just defining it once is enough.

Reply:
The way I was looking at it,  the parameters are different. Therefore the messaging will need to be different depending on if your entry point is `init.pp` vs `katello::pulp`. Otherwise the message could be confusing since the parameters have different names.

Reply:
I'd say the explicit parameters do not need to be mentioned. If you think about the installer context, it'll be confusing that you specified `--katello-expose-mongodb` but then the error talks about `expose_mongodb`. That's why I'd phrase it more generic. Something like:

Exposing MongoDB is only possible on managed databases. Either manage the database or do not expose.